fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/matcher/DefaultMatcherOperatorRegistry.java

```diff
@@ -0,0 +1,84 @@
+package com.navercorp.fixturemonkey.api.matcher;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import com.navercorp.fixturemonkey.api.property.Property;
+import com.navercorp.fixturemonkey.api.type.Types;
+
+public class DefaultMatcherOperatorRegistry<T> implements MatcherOperatorRegistry<T> {
```

Comment:
There is no need to inherit right now; please add `final`.

Reply:
I think the `MatcherOperatorRegistry` splits two concerns: the registry and the retriever.

The `MatcherOperatorRegistry` should have the methods `addFirst` and `addLast`.

A new interface is needed to get the `List`.

The implementations of two interface should be `DefaultMatcherOperatorRegistry`

Reply:
I agree with your idea of separating concerns into a Retriever and a Registry.  
The original `DefaultMatcherOperatorRegistry` was responsible for both concerns,  
so I thought it would be clearer to rename it to `Container`.